src/adapters/ws-server/index.ts

```diff
@@ -0,0 +1,148 @@
+import WebSocket from 'ws'
+import http from 'http'
+import { validateData } from '../../lib/util.js'
+import Adapter from '../../lib/adapter.js'
+import GleeConnection from '../../lib/connection.js'
+import GleeMessage from '../../lib/message.js'
+import GleeError from '../../errors/glee-error.js'
+
+class WebSocketsAdapter extends Adapter {
+  name(): string {
+    return 'WebSockets adapter'
+  }
+
+  async connect(): Promise<this> {
+    return this._connect()
+  }
+
+  async send(message: GleeMessage): Promise<void> {
+    return this._send(message)
+  }
+
+  _connect(): Promise<this> {
+    return new Promise((resolve, reject) => {
+      const serverUrl = new URL(this.serverUrlExpanded)
+      const wsHttpServer = this.glee.options?.websocket?.httpServer || http.createServer()
```

Comment:
Since we are creating ws client I think we should do something like this 
```suggestion
      const wsClient = new Websocket(serverUrl);
```

Reply:
This is the WS server but you're right, we should be doing this on the client, which is in the `ws-client-/indext.ts` file.

Reply:
Actually, I was not able to comment on the `ws-client/index.ts` file. Sorry for any confusion.